src/common/apple/purchase.ts

```diff
@@ -0,0 +1,247 @@
+} from '../utils';
+import type { Block, KnownBlock } from '@slack/web-api';
+import { webhooks } from '../slack';
+import { checkUserCoresAccess } from '../user';
+import { CoresRole } from '../../types';
+
+export const isCorePurchaseApple = ({
+  transactionInfo,
+}: {
+  transactionInfo: JWSTransactionDecodedPayload;
+}) => {
+  return (
+    getAppleTransactionType({ transactionInfo }) ===
+      AppleTransactionType.Consumable &&
+    !!transactionInfo.productId?.startsWith('cores_')
+  );
+};
+
+export const notifyNewStoreKitPurchase = async ({
+  data,
+  transaction,
+  user,
+  currencyInUSD,
+}: {
+  data: JWSTransactionDecodedPayload;
+  transaction: UserTransaction;
+  user: Pick<User, 'id' | 'subscriptionFlags' | 'coresRole'>;
+  currencyInUSD: number;
+}) => {
+  if (isTest) {
+    return;
+  }
+
+  const blocks: (KnownBlock | Block)[] = [
+    {
+      type: 'header',
+      text: {
+        type: 'plain_text',
+        text: 'Cores purchased :cores: :apple-ico:',
+        emoji: true,
+      },
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Transaction ID:*', data.appTransactionId),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*App Account Token:*',
+            data.appAccountToken,
+          ),
+        },
+      ],
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Product:*', data.productId),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Cores:*', transaction.value.toString()),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*Purchased by:*',
+            `<https://app.daily.dev/${user.id}|${user.id}>`,
+          ),
+        },
+      ],
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*Cost:*',
+            new Intl.NumberFormat('en-US', {
+              style: 'currency',
+              currency: 'USD',
+            }).format(currencyInUSD || 0),
+          ),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Currency:*', 'USD'),
+        },
+      ],
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*Cost (local):*',
+            new Intl.NumberFormat('en-US', {
+              style: 'currency',
+              currency: data.currency,
+            }).format((data.price || 0) / 1000),
+          ),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Currency (local):*', data.currency),
+        },
+      ],
+    },
+  ];
+
+  await webhooks.transactions.send({ blocks });
+};
+
+export const handleCoresPurchase = async ({
```

Comment:
Once we approve everything this should be mostly the same as the paddle cores?
I assume the same internal logic, just different trigger?

Reply:
it is not the same since payloads differ, but `purchaseCores` is reused